rust/connlib/tunnel/src/control_protocol/client.rs

```diff
@@ -129,13 +133,19 @@ where
                     }
                 };
 
-                d.on_close(on_dc_close_handler(index, gateway_id, tunnel.stop_peer_command_sender.clone()));
+                d.on_close(on_dc_close_handler(
+                    index,
+                    gateway_id,
+                    tunnel.stop_peer_command_sender.clone(),
+                ));
+                let d = d.detach().await.expect(
+                    "only fails if not opened or not enabled, both of which are always true for us",
```

Comment:
Sorry, this is not part of this PR but here we will panic in a race condition where the channel closes while we are executing this callback. It's unlikely but I'm not sure it's impossible.

Reply:
Do we care?

Reply:
I do think we should at least not panic and remove the peer.

Reply:
Where do you see this race condition? The source only mentions an error if it is called before opening: https://docs.rs/webrtc/latest/src/webrtc/data_channel/mod.rs.html#400-413

Reply:
I cannot find any place in the source code where this is set back to `None` after it has been opened.

Reply:
My bad, I thought that it might have set to `None` after closing.